lib/libc/minimal/source/stdlib/rand.c

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2021 Space Cubics, LLC.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdlib.h>
+#include <stdint.h>
+#include <toolchain.h>
+
+#define OUTPUT_BITS (0x7fffffffU)
+#define MULTIPLIER (1103515245U)
+#define INCREMENT (12345U)
+
+static uint32_t __seed;
```

Comment:
Another question: given that the function is defined in terms of standard C types (int/unsigned), then why uint32_t is used here?

Reply:
Final question: how rand() will behave, if srand() is not explicitly called, given such seed variable definition, and comment below `/* seed of 0 doesn't generate a good random number */` ?

Reply:
for `uint32_t`, it was just my habit to use size specific type when ever possible.  I'll change it to `unsigned int` if required.

Because the seed is static, it's `0` if `srand()` is not called.  Thus, it returns `1103527590`.

Reply:
oops, `rand()` without calling `srand()` means that `rand()` will use `0` as the seed and returns `12345` instead of `1103527590` (which is the value for the seed `1`).

hmm... I'm not sure we should guard seed of 0 or not.  Does anyone have any idea?

Reply:
> Does anyone have any idea?

Yes, it should be something like:

```
 static unsigned long srand_seed = 1; 
```

(Fixing also other issues like double-underscore prefix; variable type is based on sample implementation in https://pubs.opengroup.org/onlinepubs/9699919799/functions/rand.html)

Reply:
fixed.